src/xrpld/ledger/detail/View.cpp

```diff
@@ -287,8 +311,9 @@ accountHolds(
         amount.clear(Issue{currency, issuer});
     }
     else if (
-        (zeroIfFrozen == fhZERO_IF_FROZEN) &&
-        isFrozen(view, account, currency, issuer))
+        ((zeroIfFrozen == fhZERO_IF_FROZEN) &&
+         isFrozen(view, account, currency, issuer)) ||
+        isDeepFrozen(view, account, currency, issuer))
```

Comment:
This code change, by adding an additional `isDeepFrozen` check, should be unnecessary. Namely, deep freezing a trust line is only possible when it is already regularly frozen, while having deep freeze without regular freeze is not allowed; hence, the regular freeze check is sufficient to capture both cases.

After reverting this change, the newly introduced `isDeepFreeze` function can then also be removed as it's not used anywhere else.